dvc/repo/__init__.py

```diff
@@ -149,7 +149,7 @@ def _get_pipeline(self, node):
         assert len(pipelines) == 1
         return pipelines[0]
 
-    def collect(self, target, with_deps=False):
+    def collect(self, target, with_deps=False, branch=None):
```

Comment:
is `branch` parameter used somewhere?

Reply:
Good catch! I believe it should go to `Stage.load()`. Need to write a test.

Reply:
🤔 I think `pylint` is not installed as a hook yet (black and flake8 are style linters only as far as I understand)? It is highlighted in IDE though.

Reply:
Ok, it looks like another reason to try PyCharm :)

Reply:
Btw, pylint gives some really useful hints on the DVC codebase, we should consider to spend some time on it at least (it is not reasonable to include it into CI pipeline though as of my point of view).

Reply:
@ei-grad totally! I think it's a thing @efiop is working on. I would prefer to include it actually as part of CI :)

Reply:
pylint is coming. We've done some work to comply already, but there is a lot more to do. We'll enable it as a part of CI, so that no patch can be merged without complying to it(or at least considering everything and ignoring particular warnings knowingly).